core/dbt/include/global_project/macros/etc/statement.sql

```diff
@@ -1,13 +1,27 @@
-{% macro statement(name=None, fetch_result=False, auto_begin=True) -%}
+{#--
+TODO:  Only certian adapters need to support python models so this should probably be
+moved to an override I.E. `spark__statement`  but for some reason it wasn't picking up that change
```

Comment:
The macro override naming method (spark__statement) only works for macros which are called with adapter.dispatch. For macros called directly, you can just redefine them. (macro statement)

Reply:
Though maybe not in adapters... redefining macros in a project works...

We probably need some other way to check whether python models are supported for this adapter. Possibly a method in the adapter class in impl.py?

Reply:
Gotcha, yes, I think we are going to go with Jeremy's suggested way of updating materialization to specify supported language, and default to `sql` to make restrict this